src/DIRAC/WorkloadManagementSystem/Optimizer/InputData.py

```diff
@@ -0,0 +1,431 @@
+        :param list inputData: list of LFNs
+
+        :returns: S_OK/S_ERROR structure with resolved input data info
+        """
+        lfns = inputData
+
+        result = jobState.getManifest()
+        if not result["OK"]:
+            self.jobLog.error("Failed to get job manifest", result["Message"])
+            return result
+        manifest = result["Value"]
+        vo = manifest.getOption("VirtualOrganization")
+        startTime = time.time()
+        dm = self.__getDataManager(vo)
+        if dm is None:
+            return S_ERROR("Failed to instantiate DataManager for vo %s" % vo)
+        else:
+            # This will return already active replicas, excluding banned SEs, and
+            # removing tape replicas if there are disk replicas
+            result = dm.getReplicasForJobs(lfns)
+        self.jobLog.verbose("Catalog replicas lookup time", "%.2f seconds " % (time.time() - startTime))
+        if not result["OK"]:
+            self.log.warn(result["Message"])
+            return result
+
+        replicaDict = result["Value"]
+
+        self.jobLog.verbose("REPLICA DICT", replicaDict)
+
+        result = self.__checkReplicas(replicaDict, vo)
+
+        if not result["OK"]:
+            self.jobLog.error("Failed to check replicas", result["Message"])
+            return result
+        okReplicas = result["Value"]
+
+        result = self.__getSiteCandidates(okReplicas, vo)
+        if not result["OK"]:
+            self.jobLog.error("Failed to check SiteCandidates", result["Message"])
+            return result
+        siteCandidates = result["Value"]
+
+        if self.ex_getOption("CheckFileMetadata", True):
+            result = jobState.getManifest()
+            if not result["OK"]:
+                return result
+            manifest = result["Value"]
+            vo = manifest.getOption("VirtualOrganization")
+            fc = self.__getFileCatalog(vo)
+            if fc is None:
+                return S_ERROR("Failed to instantiate FileCatalog for vo %s" % vo)
+            else:
+                guidDict = fc.getFileMetadata(lfns)
+            self.jobLog.info("Catalog Metadata Lookup Time", "%.2f seconds " % (time.time() - startTime))
+
+            if not guidDict["OK"]:
+                self.log.warn(guidDict["Message"])
+                return guidDict
+
+            failed = guidDict["Value"]["Failed"]
+            if failed:
+                self.log.warn("Failed to establish some GUIDs")
+                self.log.warn(failed)
+
+            for lfn in replicaDict["Successful"]:
+                replicas = replicaDict["Successful"][lfn]
+                guidDict["Value"]["Successful"][lfn].update(replicas)
+
+        resolvedData = {}
+        resolvedData["Value"] = guidDict
+        resolvedData["SiteCandidates"] = siteCandidates
+        self.jobLog.debug("Storing:\n%s" % pprint.pformat(resolvedData))
+        result = self.storeOptimizerParam(self.ex_getProperty("optimizerName"), resolvedData)
```

Comment:
This should also be removed.

Reply:
After looking a little bit more at InputData, I think this optimizer just do nothing at all. If you could double-check that it has absolutely no utility, then I will just get rid of it

Reply:
It does something: it checks that the input data requested have replicas.

Reply:
Indeed, thanks